src/hooks/useApproveCallback/useApproveCallbackMod.ts

```diff
@@ -79,8 +80,20 @@ export function useApproveCallback({
     // we might not have enough data to know whether or not we need to approve
     if (!currentAllowance) return ApprovalState.UNKNOWN
 
+    const notEnoughAllowance = currentAllowance.lessThan(amountToCheckAgainstAllowance || amountToApprove)
+
+    // Handle Special case for USDT and other tokens that require setting the allowance to zero before setting it to a new value
+    // TODO: Clean this
+    if (
+      notEnoughAllowance &&
+      currentAllowance.currency.address === '0xdac17f958d2ee523a2206206994597c13d831ec7' && // USDT
```

Comment:
I brought up this issue in the backend sync and @nlordell mentioned that there are more tokens than just `USDT` that behave like this.
One way to handle this in a more general way would be to first simulate `approve(some_amount)` and only if that fails try `approve(0); approve(some_amount)`.

Reply:
Exactly what @shoom3301 proposed ;)